### DISCUSSION TITLE
pyOptSparseDriver - OptView

### DISCUSSION THREAD

**Discussion by ddiama** [OpenMDAO/OpenMDAO#2987] [Q&A] [answered]
Hello, 

Is there a way to use the OptView functionality while running the optimization through the pyOptSparseDriver in OpenMDAO? 
How can I save the history file through the pyOptSparseDriver so I can visualize the optimization convergence? 

Thank you,
Dimitra

**Comment by johnjasa** [comment, marked as answer]
Hi Dimitra! Yes, this is definitely possible. Here is a short script that shows how to add the `hist_file` argument to a `pyOptSparseDriver`:

```
import openmdao.api as om

# build the model
prob = om.Problem()

prob.model.add_subsystem('paraboloid', om.ExecComp('f = (x-3)**2 + x*y + (y+4)**2 - 3'))

prob.driver = om.pyOptSparseDriver(optimizer='SNOPT')
prob.driver.hist_file = 'opt.hst'

prob.model.add_design_var('paraboloid.x', lower=-50, upper=50)
prob.model.add_design_var('paraboloid.y', lower=-50, upper=50)
prob.model.add_objective('paraboloid.f')

prob.setup()

# Set initial values.
prob.set_val('paraboloid.x', 3.0)
prob.set_val('paraboloid.y', -4.0)

# run the optimization
prob.run_driver()
```

You can then run `OptView` like normal and see an interactive window like the attached screenshot shows.

<img width="833" alt="optview" src="https://github.com/OpenMDAO/OpenMDAO/assets/16373529/5fe499bc-a9c7-4ea3-b72e-6d4f087ed09c">

I've realized our docs might be lacking this info so I've added an [OpenMDAO issue](https://github.com/OpenMDAO/OpenMDAO/issues/2988) so we can improve them.

Thank you for your question!

**Reply by johnjasa** [reply]
I should also mention the [Case Viewer](https://openmdao.org/newdocs/versions/latest/features/model_visualization/case_viewer_driver_cases.html) which is a quite relevant and helpful visualization tool built-in to OpenMDAO!

**Reply by ddiama** [reply]
Thank you very much, I was not aware of the prob.driver.hist_file option!
